libs/domain/site/menu-item/menu-item.component.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { fixture } from '@open-wc/testing-helpers';
+import { createInjector, destroyInjector } from '@spryker-oryx/di';
+import { RouterService } from '@spryker-oryx/router';
+import { LinkService } from '@spryker-oryx/site';
+import { ButtonComponent } from '@spryker-oryx/ui/button';
+import { useComponent } from '@spryker-oryx/utilities';
+import { html } from 'lit';
+import { of } from 'rxjs';
+import { SiteMenuItemComponent } from './menu-item.component';
+import { siteMenuItemComponent } from './menu-item.def';
+import { SiteMenuItemVariation } from './menu-item.model';
+
+class MockLinkService implements Partial<LinkService> {
+  get = vi.fn().mockReturnValue(of('/'));
+}
+
+class MockRouterService implements Partial<RouterService> {
+  isCurrentRoute = vi.fn().mockReturnValue(of(false));
+}
+
+describe('SiteMenuItemComponent', () => {
+  let element: SiteMenuItemComponent;
+  let routerService: MockRouterService;
+  let linkService: MockLinkService;
+
+  beforeAll(async () => {
+    await useComponent(siteMenuItemComponent);
+  });
+
+  beforeEach(async () => {
+    const testInjector = createInjector({
+      providers: [
+        {
+          provide: LinkService,
+          useClass: MockLinkService,
+        },
+        {
+          provide: RouterService,
+          useClass: MockRouterService,
+        },
+      ],
+    });
+
+    routerService = testInjector.inject(
+      RouterService
+    ) as unknown as MockRouterService;
+    linkService = testInjector.inject(
+      LinkService
+    ) as unknown as MockLinkService;
+
+    element = await fixture(
+      html`<oryx-site-menu-item
+        .content=${{ text: 'mock' }}
+      ></oryx-site-menu-item>`
+    );
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    destroyInjector();
+  });
+
+  it('should be defined', () => {
+    expect(element).toBeInstanceOf(SiteMenuItemComponent);
+  });
+
+  it('should pass the a11y audit', async () => {
+    await expect(element).shadowDom.to.be.accessible();
+  });
+
+  it('should render its components', () => {
+    expect(element).toContainElement('oryx-button');
+  });
+
+  it('should pass text to oryx button', () => {
+    expect(
+      element.renderRoot.querySelector<ButtonComponent>('oryx-button')?.text
+    ).toBe('mock');
+  });
+
+  it('should not be highlighted', () => {
+    expect(element).not.toContainElement('.active');
```

Comment:
Isn't it better to specify that button shouldn't be active?
```suggestion
    expect(element).not.toContainElement('oryx-button.active');
```